src/main/java/com/google/api/gax/grpc/RetryingCallable.java

```diff
@@ -29,198 +29,62 @@
-import com.google.api.gax.core.ApiFutureCallback;
-import com.google.api.gax.core.ApiFutures;
+import com.google.api.gax.core.NanoClock;
 import com.google.api.gax.core.RetrySettings;
+import com.google.api.gax.core.internal.ApiFutureToListenableFuture;
+import com.google.api.gax.retrying.RetryAttemptSettings;
+import com.google.api.gax.retrying.RetryFuture;
+import com.google.api.gax.retrying.ScheduledRetryHandler;
 import com.google.common.base.Preconditions;
 import io.grpc.CallOptions;
-import io.grpc.Status;
-import java.util.concurrent.CancellationException;
-import java.util.concurrent.Future;
-import java.util.concurrent.ThreadLocalRandom;
+import io.grpc.Status.Code;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
-import javax.annotation.Nullable;
 import org.joda.time.Duration;
 
 /**
- * Implements the retry and timeout functionality used in {@link UnaryCallable}. The behavior is
- * controlled by the given {@link RetrySettings}.
+ * Implements the retry and timeout functionality used in {@link UnaryCallable}.
+ *
+ * The behavior is controlled by the given {@link RetrySettings}.
  */
 class RetryingCallable<RequestT, ResponseT> implements FutureCallable<RequestT, ResponseT> {
-
   // Duration to sleep on if the error is DEADLINE_EXCEEDED.
   static final Duration DEADLINE_SLEEP_DURATION = Duration.millis(1);
 
   private final FutureCallable<RequestT, ResponseT> callable;
   private final RetrySettings retryParams;
-  private final UnaryCallable.Scheduler executor;
+  private final ScheduledExecutorService scheduler;
   private final NanoClock clock;
 
   RetryingCallable(
       FutureCallable<RequestT, ResponseT> callable,
       RetrySettings retrySettings,
-      UnaryCallable.Scheduler executor,
+      ScheduledExecutorService scheduler,
       NanoClock clock) {
     this.callable = Preconditions.checkNotNull(callable);
     this.retryParams = Preconditions.checkNotNull(retrySettings);
-    this.executor = executor;
+    this.scheduler = scheduler;
     this.clock = clock;
   }
 
   @Override
   public ApiFuture<ResponseT> futureCall(RequestT request, CallContext context) {
-    RetryingResultFuture resultFuture = new RetryingResultFuture();
-    context = getCallContextWithDeadlineAfter(context, retryParams.getTotalTimeout());
-    Retryer retryer =
-        new Retryer(
-            request,
-            context,
-            resultFuture,
-            retryParams.getInitialRetryDelay(),
-            retryParams.getInitialRpcTimeout(),
-            null);
-    resultFuture.issueCall(request, context, retryer);
-    return resultFuture;
+    GrpcRetryCallable<RequestT, ResponseT> retryCallable =
+        new GrpcRetryCallable<>(callable, request, context);
+    GrpcRetryHandler<ResponseT> retryHelper = new GrpcRetryHandler<>(clock, scheduler);
+    RetryFuture<ResponseT> rv = retryHelper.createFirstAttempt(retryCallable, retryParams);
```

Comment:
Ok. rv means "returned value", I use it often, but I agree, that it is rather a bad habit than a good practice.